Dockerfile

```diff
@@ -1,3 +1,7 @@
 FROM nginx:1.15-alpine
 COPY dist/hypertrace-ui /usr/share/nginx/html
-COPY conf/default.conf /etc/nginx/conf.d/default.conf
\ No newline at end of file
+COPY conf/default.conf /etc/nginx/conf.d/default.conf
+# expose port for UI
+EXPOSE 2020
+# Healthcheck to check if client can connect to UI port
+HEALTHCHECK --interval=1s --retries=5 --timeout=1s CMD wget -qO- http://localhost:2020/graphql\?query\=\{metadata\{name\}\} &>/dev/null || exit 1
```

Comment:
@JBAhire I verified the previous query http://localhost:2020/ query using developer console, and it's not heavy. I think this will be slow the first time. So, can we revert back to previous health check only? Sorry for the confusion here.

Reply:
@kotharironak and @ravisingal , so we are not doing any deep healthcheck here?

Reply:
@kotharironak 's initial concern seems valid to me, but maybe I'm missing some context. Not super familiar with docker's impl of healthcheck, but I read this as trying to load the default dashboard every 1s?

Reply:
I do agree that pulling `/` is not ideal, but I also think that faking the test with netcat is worse than no healthcheck, as nginx will rarely fail to start. We need to understand why `http://localhost:2020/graphql\?query\=\{metadata\{name\}\}` specifically is bad, and if it is only about first time long request, I don't think that's an issue as a start period already will wash that out.

@kotharironak I think you agree this is not the same as hitting `/`, so can we get to the bottom of  this and merge something that isn't a fake? We've all spent a lot of time on this, so doing a last 5% seems worth it vs punting to netcat.